### ISSUE TITLE
Date of last high risk contact wrong

### ISSUE THREAD

**Issue by masema67** [corona-warn-app/cwa-documentation#818]
## Avoid duplicates
* [X ] Bug is not mentioned in the [FAQ](https://www.coronawarn.app/en/faq/)

## Technical details

Device name: Google Pixel 4a 5G
iOS/Android version: Android 12
App version: 2.16.2

## Describe the bug

positive rapid test and positive PCR  test conducted and registered in CWA at the 20.01.2022

People with very close contact to this person for hours and every day receive high risks warnings for the 14.01 to 17.01 but not for the 18.01, 19.01, 20.01 even so in very close contact to the person for hours at those dates.

## Steps to reproduce the issue
see above

## Expected behaviour

last contact should be displayed for the 20.01 and not 17.01

**Comment by Ein-Tim** [comment]
The dates come directly from the underlying OS framework.

Seems like on days 18.01-20.01 CWA sees no contacts "above" the threshold.

More information: https://www.coronawarn.app/en/faq/#encounter_19_calc

**Comment by MikeMcC399** [comment]
@masema67 

There is no complete up-to-date description of the risk calculation process. You might like to refer though to https://github.com/corona-warn-app/cwa-documentation/blob/master/cwa-risk-assessment.md which shows how the transmission risk level decreases over time. The app also takes account if / when symptoms were reported. The collection of FAQ articles under https://www.coronawarn.app/en/faq/#risk_assessment also discusses some aspects.

So what you describe could be correct. It is impossible to check for an individual case though because access to the data used in the calculations is restricted, due to privacy protections.

**Comment by masema67** [comment]
> The dates come directly from the underlying OS framework.
> 
> Seems like on days 18.01-20.01 CWA sees no contacts "above" the threshold.
> 
> More information: https://www.coronawarn.app/en/faq/#encounter_19_calc


""above" the threshold" is for sure not possible, as contacts where for hours with partly no distance between the phones for quite some time

**Comment by masema67** [comment]
> @masema67
> 
> There is no complete up-to-date description of the risk calculation process. You might like to refer though to https://github.com/corona-warn-app/cwa-documentation/blob/master/cwa-risk-assessment.md which shows how the transmission risk level decreases over time. The app also takes account if / when symptoms were reported. The collection of FAQ articles under https://www.coronawarn.app/en/faq/#risk_assessment also discusses some aspects.
> 
> So what you describe could be correct. It is impossible to check for an individual case though because access to the data used in the calculations is restricted, due to privacy protections.

Thank you! The same behavior is observed with CCTG - so it is not google but for sure a not sense making behavior by CWA.

**Comment by dsarkar** [comment]
Dear @masema67,

Thank you very much for this report. We have been discussing this case internally, and yes, the indications, although surprising, could be correct as you describe it. However, we would be very interested to talk to you, in order to get a more detailed insight about this specific case. 

A developer/architect from the team would talk to you, ideally by telephone. Please, contact us via email (corona-warn-app.opensource@sap.com),  in order to arrange details. Please mention in the email the Internal Tracking ID: [EXPOSUREAPP-11569](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11569)



Best wishes, DS

---
Corona-Warn-App Open Source Team

**Comment by mlenkeit** [comment]
> There is no complete up-to-date description of the risk calculation process. 

@MikeMcC399 what about https://github.com/corona-warn-app/cwa-documentation/blob/master/solution_architecture.md#risk-calculation?

**Comment by MikeMcC399** [comment]
@mlenkeit 

>>There is no complete up-to-date description of the risk calculation process.
>
>what about https://github.com/corona-warn-app/cwa-documentation/blob/master/solution_architecture.md#risk-calculation?

Thanks for reminding us about the section in the Solution Architecture document. This at least contains the description of how Onset of Symptoms is used.

Could you please check about the Figure 16: Risk calculation
https://github.com/corona-warn-app/cwa-documentation/blob/master/images/solution_architecture/risk_calculation.svg

which has a threshold of 13 minutes (where I would expect this to show 9 minutes).

According to https://www.coronawarn.app/en/blog/2021-04-16-corona-warn-app-risk-calculation-further-improved/ dated April 16, 2021 the risk calculation was changed last year:

"For app users, this means: Previously, an encounter between two people had to last at least 13 minutes in order to be considered for an encounter with an increased risk (red tile). Now, the app considers encounters of at least 9 minutes. Thus, encounters that represent an actual relevant contact with COVID-19 infected persons can be detected even more accurately and represented accordingly in the app as a warning about an encounter with increased risk"

So the diagram and the blog information are contradictory.

- There is a related open issue https://github.com/corona-warn-app/cwa-documentation/issues/586 from April 2021 which has yet to be resolved.

**Comment by MikeMcC399** [comment]
@mlenkeit 

I took a closer look at the diagram https://github.com/corona-warn-app/cwa-documentation/blob/master/images/solution_architecture/risk_calculation.svg

On the left-hand side it is still using 13 minutes and on the right-hand side it uses 9 minutes.

**Comment by masema67** [comment]
> Dear @masema67,
> 
> Thank you very much for this report. We have been discussing this case internally, and yes, the indications, although surprising, could be correct as you describe it. However, we would be very interested to talk to you, in order to get a more detailed insight about this specific case.
> 
> A developer/architect from the team would talk to you, ideally by telephone. Please, contact us via email ([corona-warn-app.opensource@sap.com](mailto:corona-warn-app.opensource@sap.com)), in order to arrange details. Please mention in the email the Internal Tracking ID: [EXPOSUREAPP-11569](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-11569)
> 
> Best wishes, DS
> 
> Corona-Warn-App Open Source Team

I sent them an email and will be glad to give more details in the phone.
best
masema67

**Comment by dsarkar** [comment]
@masema67 Ok! Thanks!

Best wishes, DS

---
Corona-Warn-App Open Source Team

**Comment by MikeMcC399** [comment]
The risk calculation diagram [images/solution_architecture/risk_calculation.svg](https://github.com/corona-warn-app/cwa-documentation/blob/master/images/solution_architecture/risk_calculation.svg) has now been corrected.

**Comment by thomasaugsten** [comment]
Thanks for updating the date of symptoms onset. Only 4 days after the onset leads to TRL > 2 which shows a warning to other persons
<img width="580" alt="image" src="https://user-images.githubusercontent.com/6367738/151149403-06469945-116f-4f78-b7f6-0ed7858fe904.png">

**Comment by MikeMcC399** [comment]
@thomasaugsten 
Thanks for the post with the diagram. This is from the presentation https://www.coronawarn.app/assets/documents/cwa-rc3.pdf from @tklingbeil in Dec 2020 (see https://www.coronawarn.app/en/blog/2020-12-30-cwa-behind-the-scenes/ for anybody who would like to view the video as well).

---

I see that @masema67 updated the information. Onset of symptoms Jan 14, 2022 together with test date Jan 20, 2022. By my calculation, onset of symptoms is 6 days before submission.

Using the table from the [presentation](https://www.coronawarn.app/assets/documents/cwa-rc3.pdf ) on page 15, with the row 6 "Days since the onset of symptoms (at the point of submission)", Day 1 (Jan 19) and Day 0 (Jan 20) are both shown with roman numeral "II" Transmission Risk Level (TRL).

The updated diagram of risk calculation on [images/solution_architecture/risk_calculation.svg](https://github.com/corona-warn-app/cwa-documentation/blob/master/images/solution_architecture/risk_calculation.svg) shows TRL of "II" being weighted with a factor 0.0. This means that exposures on the days Jan 19 and Jan 20 do not result in a red risk result and therefore the behavior which @masema67 reported as an issue, seems to be correct.

Maybe @thomasaugsten could confirm that the above is correct?

**Comment by mlenkeit** [comment]
@MikeMcC399 yes, that's correct. The 19th and 20th both translate to TRL <= II, and thus, do not count as exposures and are rightly not considered for the date of the last exposure.

**Comment by MikeMcC399** [comment]
@masema67 

It seems that we have a good explanation for what you saw. If you agree that your issue has been addressed, you could close the issue here.

**Comment by ndegendogo** [comment]
@mlenkeit I assume / hope you are in close feedback loop with RKI if the "risk profile" is still considered good from today's point of view?
Iirc it was published in Dec 2020 (but maybe the numbers were updated since then?)
Since then we had Alpha, Delta, and now Omikron.

**Comment by MikeMcC399** [comment]
@ndegendogo 
There was a blog post on this subject a couple of weeks ago. See [Wird die CWA an die Omikron-Variante angepasst?](https://www.coronawarn.app/en/blog/2022-01-11-cwa-omikron/). That wasn't really the subject of this issue however. 

- The discussion about Omikron is in https://github.com/corona-warn-app/cwa-documentation/issues/787

**Comment by mlenkeit** [comment]
@ndegendogo the blog post that @MikeMcC399 referenced (thx!) sums it up pretty well 😊

**Comment by MikeMcC399** [comment]
Is there anything further which can be done on this issue?

**Comment by larswmh** [comment]
Closing as suggested.
